### PR TITLE
feat(rancher): add tracking

### PR DESCRIPTION
**Pull request by lionel95200x**
| Question         | Answer
| ---------------- | ---
| Branch?          | `epic` 
| Bug fix?         | no
| New feature?     | no
| Breaking change? | no
| Tickets          | 12791
| License          | BSD 3-Clause

- [x] Try to keep pull requests small so they can be easily reviewed.
- [x] Commits are signed-off
- [x] Only FR translations have been updated
- [ ] Branch is up-to-date with target branch
- [ ] Lint has passed locally
- [ ] Standalone app was ran and tested locally
- [ ] Ticket reference is mentioned in linked commits (internal only)
- [ ] Breaking change is mentioned in relevant commits

### CODE REVIEW

File: packages/manager/apps/pci-rancher/src/hooks/useTrackingPage.ts [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+  TrackingPageView,
+} from '@/utils/tracking';
+
+export const useTrackingPage = (page?: TrackingPageView) => {
+  const tracking = useTracking();
+
+  useEffect(() => {
+    tracking.trackPage({
+      name: `${TRACKING_PATH}${page}`,
+      level2: 'Rancher',
```
Comment (aboungnaseng-ovhcloud) [inline packages/manager/apps/pci-rancher/src/hooks/useTrackingPage.ts:15]: Im not sure for this

Comment (aboungnaseng-ovhcloud) [reply packages/manager/apps/pci-rancher/src/hooks/useTrackingPage.ts:15]: 86 for pci

File: packages/manager/apps/pci-rancher/src/components/Modal/GenerateAccesModal.tsx [hunk 2]
```diff
@@ -79,7 +83,15 @@ const GenerateAccessModal: FC<GenerateAccessModalProps> = ({
         slot="actions"
         variant={ODS_BUTTON_VARIANT.stroked}
         color={ODS_THEME_COLOR_INTENT.primary}
-        onClick={() => toggleModal(false)}
+        onClick={() => {
+          trackAction(
+            TrackingPageView.GenerateAccessModal,
+            hasValidAccess
+              ? 'generate-access::close'
+              : 'generate-access::cancel',
```
Comment (ghyenne) [inline packages/manager/apps/pci-rancher/src/components/Modal/GenerateAccesModal.tsx:91]: Maybe you can do as other place put a constante in constante file.

File: packages/manager/apps/pci-rancher/src/components/Modal/GenerateAccesModal.tsx [hunk 3]
```diff
@@ -91,8 +103,22 @@ const GenerateAccessModal: FC<GenerateAccessModalProps> = ({
           ? {
               target: OdsHTMLAnchorElementTarget._blank,
               href: rancher.currentState.url,
+              onClick: () => {
+                trackAction(
+                  TrackingPageView.GenerateAccessModal,
+                  'generate-access::confirm',
```
Comment (ghyenne) [inline packages/manager/apps/pci-rancher/src/components/Modal/GenerateAccesModal.tsx:109]: same https://github.com/ovh/manager/pull/10964#discussion_r1495510750

File: packages/manager/apps/pci-rancher/src/components/Modal/GenerateAccesModal.tsx [hunk 4]
```diff
@@ -91,8 +103,22 @@ const GenerateAccessModal: FC<GenerateAccessModalProps> = ({
...
+                );
+              },
             }
-          : { onClick: onGenerateAccess })}
+          : {
+              onClick: () => {
+                trackAction(
+                  TrackingPageView.GenerateAccessModal,
+                  'generate-access::access-ui',
```
Comment (ghyenne) [inline packages/manager/apps/pci-rancher/src/components/Modal/GenerateAccesModal.tsx:117]: same https://github.com/ovh/manager/pull/10964#discussion_r1495510750

**Review by ghyenne — APPROVED** [review]

**Review by aboungnaseng-ovhcloud — APPROVED** [review]

**Pull request merged**